priam/src/main/java/com/netflix/priam/utils/TokenManager.java

```diff
@@ -45,9 +72,10 @@
          * unit test failures.
          */
         Preconditions.checkArgument(position >= 0, "position must be >= 0");
-        return MAXIMUM_TOKEN.divide(BigInteger.valueOf(size))
+        return tokenRangeSize.divide(BigInteger.valueOf(size))
```

Comment:
In case of RandomPartitioner, tokenRangeSize == MAXIMUM_TOKEN.
I agree that strictly speaking there's not feature toggle, but the logic looks safe for me.